callbacks/callmethod.go

```diff
@@ -13,11 +13,15 @@ func callMethod(db *gorm.DB, fc func(value interface{}, tx *gorm.DB) bool) {
 		case reflect.Slice, reflect.Array:
 			db.Statement.CurDestIndex = 0
 			for i := 0; i < db.Statement.ReflectValue.Len(); i++ {
-				fc(reflect.Indirect(db.Statement.ReflectValue.Index(i)).Addr().Interface(), tx)
+				if value := reflect.Indirect(db.Statement.ReflectValue.Index(i)); value.CanAddr() {
+					fc(value.Addr().Interface(), tx)
+				}
```

Comment:
Should we raise an error if the value is not addressable? it should be ok to return an error in those cases would panic before.

Reply:
`err = DB.Model(Product5{}).Where("id", p.ID).Update("name", "update_name_2").Error`

If so, this usage will raise an error, I'm not sure if it's good, it's up to you.

Reply:
> `err = DB.Model(Product5{}).Where("id", p.ID).Update("name", "update_name_2").Error`
> 
> If so, this usage will raise an error, I'm not sure if it's good, it's up to you.

I thought I sent this message yesterday, but in fact it was pending.... 😨